File: react/src/services/PeopleService.js
import axios from 'axios'



const PEOPLE_BASE_API_URL='http://localhost:1312/api/v1/people';
class PeopleService{

    getPeoples(){
        return axios.get(PEOPLE_BASE_API_URL);
    }

    createPeople(peopleDto){
        return axios.post(PEOPLE_BASE_API_URL,peopleDto);
    }

    getPeopleById(peoID){
        
        return axios.get(PEOPLE_BASE_API_URL + '/' + peoID);
    }
    updatePeople(peopleDto, peoID){
        return axios.put(PEOPLE_BASE_API_URL + '/' + peoID, peopleDto);
    }

    deletePeople(peoID){
        
        return axios.delete(PEOPLE_BASE_API_URL + '/' + peoID );
    }

}
export default new PeopleService()

File: src/main/java/com/fulls/fullstack/service/PeopleService.java
package com.fulls.fullstack.service;

import com.fulls.fullstack.dto.PeopleDTO;
import com.fulls.fullstack.model.People;

import java.util.List;

public interface PeopleService {

    People createPeople(PeopleDTO peopleDTO);
    People updatePeople( long peoID,PeopleDTO peopleDTO);
    People getPeople(long peoID);
    List<People> listPeople();
    void deletePeople(long peoID);

}
